subsys/net/lib/sockets/sockets.c

```diff
@@ -1337,13 +1385,19 @@ static inline ssize_t zsock_recv_dgram(struct net_context *ctx,
 	if (src_addr && addrlen) {
 		if (IS_ENABLED(CONFIG_NET_OFFLOAD) &&
 		    net_if_is_ip_offloaded(net_context_get_iface(ctx))) {
-			/*
-			 * Packets from offloaded IP stack do not have IP
-			 * headers, so src address cannot be figured out at this
-			 * point. The best we can do is returning remote address
-			 * if that was set using connect() call.
-			 */
-			if (ctx->flags & NET_CONTEXT_REMOTE_ADDR_SET) {
+			if (!net_pkt_addr_is_unspecified(pkt)) {
+				int rv;
+
+				rv = sock_get_offload_pkt_src_addr(pkt,
+							    src_addr, *addrlen);
+
+				if (rv < 0) {
+					errno = -rv;
+					LOG_ERR("sock_get_offload_pkt_src_addr %d",
```

Comment:
Please change this to `NET_DBG()`, the error log is probably overkill.
I see that you copied this from few lines below, I would not mind if that code is changed too (meaning rv->ret, and LOG_ERR -> NET_DBG) but perhaps in separate commit.

Reply:
I have renamed to` NET_DBG` and `ret`. If there is no problem i think we can  push  this in this commit.